r/src/safe-call-into-r.h

```diff
@@ -93,7 +100,8 @@ MainRThread& GetMainRThread();
 // a SEXP (use cpp11::as_cpp<T> to convert it to a C++ type inside
 // `fun`).
 template <typename T>
-arrow::Future<T> SafeCallIntoRAsync(std::function<arrow::Result<T>(void)> fun) {
+arrow::Future<T> SafeCallIntoRAsync(std::function<arrow::Result<T>(void)> fun,
+                                    std::string reason = "unspecified") {
```

Comment:
Not related to this PR, but is it posible to properly propagate the R error below instead of some placeholder `UnknownError` (other languages such as Python and Java try to do that)?

One possibility is to use `StatusDetail` to keep as much R information as desired (which may also allow to roundtrip later on, i.e. re-throw the R error to the R interpreter from an outer level).

Reply:
Or perhaps the `SetError(e.token)` ensures that the error is kept around? But then `SafeCallIntoR` should not be called directly?

Reply:
Right, this needn't be done here (and thanks for working on all this :-)).